From 0ebb59b83d4024d62e398b557d2924fe5022313b Mon Sep 17 00:00:00 2001
From: Dave Stevenson <dave.stevenson@raspberrypi.com>
Date: Fri, 21 Oct 2022 14:27:45 +0100
Subject: [PATCH] drm/vc4: Increase number of overlay planes from 16 to 48

The HVS can accept an arbitrary number of planes, provided
that the overall pixel read load is within limits, and
the display list can fit into the dlist memory.

Now that DRM will support 64 planes per device, increase
the number of overlay planes from 16 to 48 so that the
dlist complexity can be increased (eg 4x4 video wall on
each of 3 displays).

Signed-off-by: Dave Stevenson <dave.stevenson@raspberrypi.com>
---
 drivers/gpu/drm/vc4/vc4_plane.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- a/drivers/gpu/drm/vc4/vc4_plane.c
+++ b/drivers/gpu/drm/vc4/vc4_plane.c
@@ -2610,7 +2610,7 @@ struct drm_plane *vc4_plane_init(struct
 	return plane;
 }
 
-#define VC4_NUM_OVERLAY_PLANES	16
+#define VC4_NUM_OVERLAY_PLANES	48
 
 int vc4_plane_create_additional_planes(struct drm_device *drm)
 {
